lib/cog/commands/pluck.ex

```diff
@@ -0,0 +1,112 @@
+defmodule Cog.Commands.Pluck do
+  use Spanner.GenCommand.Base, bundle: Cog.embedded_bundle, enforcing: false, calling_convention: :all
+  alias Cog.Command.Helpers, as: CommandHelpers
+
+  @moduledoc """
+  Returns the specified fields from a set of inputs.
+
+  ## Example
+
+      @bot #{Cog.embedded_bundle}:rules --list --for-command="#{Cog.embedded_bundle}:permissions" | #{Cog.embedded_bundle}:pluck --fields=id
+      > { "id": "91edb472-04cf-4bca-ba05-e51b63f26758" }
+        { "id": "738ce922-14cd-4abc-fe94-81658707229c" }
+      @bot #{Cog.embedded_bundle}:rules --list --for-command="#{Cog.embedded_bundle}:permissions" | #{Cog.embedded_bundle}:pluck --fields=rule,command
+      > { "rule": "operable:manage_users",
+          "command": "operable:permissions" }
+        { "rule": "operable:manage_users",
+          "command": "operable:permissions" }
+      @bot #{Cog.embedded_bundle}:seed '[{"foo":{"bar.qux":{"baz":"stuff"}}}, {"foo": {"bar":{"baz":"me"}}}]' | #{Cog.embedded_bundle}:pluck --fields=foo"
+      > [ {"foo": {"bar.qux": {"baz": "stuff"} } }, {"foo": {"bar": {"baz": "me"} } } ]
+      @bot #{Cog.embedded_bundle}:seed '[{"foo":{"bar":{"baz":"stuff"}}}, {"foo": {"bar":{"baz":"me"}}}]' | #{Cog.embedded_bundle}:pluck --fields="foo.bar"
+      > { "foo.bar": {"baz": "stuff"} }
+        { "foo.bar": {"baz": "me"} }
+      @bot #{Cog.embedded_bundle}:seed '[{"foo":{"bar":{"baz":"stuff"}}, "qux": "one"}, {"foo": {"bar":{"baz":"me"}}, "qux": "two"}]' | #{Cog.embedded_bundle}:pluck --fields="qux,\"foo.bar\""
+      > { "qux": "one", "foo.bar": {"baz": "stuff"} }
+        { "qux": "two", "foo.bar": {"baz": "me"} }
+  """
+
+  option "fields", type: "string", required: false
+
+  defstruct req: nil, fields: nil, input: nil, output: nil, errors: []
+
+  def handle_message(req, state) do
+    case req |> validate |> execute |> format do
```

Comment:
I get the idea, but if I'm understanding what I'm reading correctly a failure in validate would cause the flow to bypass execution anyway. So you are most likely only ever going to have a single error returned to the user. And if that wasn't the case, often times an error early on will ripple through the rest of flow causing more errors down the line. And don't get me wrong, I'm not saying what you did is wrong, but in my opinion it keeps things a lot simpler when you fail quickly.